sky/serve/replica_managers.py

```diff
@@ -918,3 +1002,28 @@ def _replica_prober(self) -> None:
                 with ux_utils.enable_traceback():
                     logger.error(f'  Traceback: {traceback.format_exc()}')
             time.sleep(serve_constants.ENDPOINT_PROBE_INTERVAL_SECONDS)
+
+    #########################################
+    # SkyServe Update and replica versioning. #
+    #########################################
+
+    def update_version(self, version: int, spec: 'service_spec.SkyServiceSpec',
+                       mixed_replica_versions: bool,
+                       task_yaml_path: str) -> None:
+        self.version2spec[version] = spec
+        self.version2yaml[version] = task_yaml_path
```

Comment:
Consider saving the data into the controller database rather than memory.

Reply:
sounds good